CodeLite/UnixProcess.cpp

```diff
@@ -78,16 +78,27 @@ bool UnixProcess::ReadAll(int fd, std::string& content, int timeoutMilliseconds)
     int ms = timeoutMilliseconds % 1000;
 
     struct timeval tv = { seconds, ms * 1000 }; //  10 milliseconds timeout
-    int rc = ::select(fd + 1, &rset, nullptr, nullptr, &tv);
-    if(rc > 0) {
-        memset(buff, 0, sizeof(buff));
-        if(read(fd, buff, (sizeof(buff) - 1)) > 0) {
-            content.append(buff);
+    while (true) {
+        int rc = ::select(fd + 1, &rset, nullptr, nullptr, &tv);
+        if(rc > 0) {
+            int len = read(fd, buff, (sizeof(buff) - 1));
+            if(len > 0) {
+                buff[len] = 0;
+                content.append(buff);
+                if (content.length() >= 2048 * 1024) {
+                    return true;
+                }
+                tv.tv_usec = 0;
+                tv.tv_sec = 0;
+                FD_ZERO(&rset);
+                FD_SET(fd, &rset);
+                continue;
+            }
+        } else if(rc == 0) {
+            // timeout
             return true;
         }
-    } else if(rc == 0) {
-        // timeout
-        return true;
+        break;
```

Comment:
this is wrong. We should call here `return true` as it was before. timeout on select is OK.
Especially when you changed the code call `select` multiple times